src/Microsoft.Azure.WebJobs.Host/RetryContext.cs

```diff
@@ -25,5 +34,10 @@ public class RetryContext
         /// Gets or sets the <see cref="IFunctionInstance"/> for the failed invocation.
         /// </summary>
         public IFunctionInstance Instance { get; set; }
+
+        /// <summary>
+        /// Gets the state dictionary.
+        /// </summary>
+        internal IDictionary<string, object> StateDictionary => _stateDictionary;
```

Comment:
Can we just name this "State"? We can also add a <remarks> doc section here to the effect "This property can be used by retry strategies to store arbitrary state for a retried invocation.".